crates/witness-generator/src/rpc_generator.rs

```diff
@@ -103,6 +103,13 @@ impl WitnessGenerator for RpcBlocksAndWitnesses {
     ///
     /// Returns either the last N blocks or a specific block with their execution witnesses.
     async fn generate(&self) -> Result<Vec<BlockAndWitness>> {
+        // If live polling is enabled, we return an error here
+        if self.stop.is_some() {
```

Comment:
Realized this by doing the tests. ha

Under the polling configuration, given is a long running call until stopped, the only supported call now is `generate_to_path` since results are constantly saved in the files.

I think in the future we can support returning the result in a `Vec` but I'm not sure this will be used by somebody, since if you're doing really long runs (as we do in our daemon VM), that would be a huge vector. But maybe for smaller time spans might make sense.